io/io/inc/ROOT/RRawFile.hxx

```diff
@@ -0,0 +1,130 @@
+// @(#)root/io:$Id$
+// Author: Jakob Blomer
+
+/*************************************************************************
+ * Copyright (C) 1995-2018, Rene Brun and Fons Rademakers.               *
+ * All rights reserved.                                                  *
+ *                                                                       *
+ * For the licensing terms see $ROOTSYS/LICENSE.                         *
+ * For the list of contributors see $ROOTSYS/README/CREDITS.             *
+ *************************************************************************/
+
+#ifndef ROOT_RRawFile
+#define ROOT_RRawFile
+
+#include <ROOT/RStringView.hxx>
+
+#include <cstddef>
+#include <cstdint>
+#include <string>
+
+namespace ROOT {
+namespace Detail {
+
+class RRawFile;
+
+/**
+ * \class RRawFile RRawFile.hxx
+ * \ingroup IO
+ *
+ * The RRawFile provides read-only access to local and remote files. Data can be read either byte-wise or line-wise.
+ * The RRawFile base class provides line-wise access and buffering for byte-wise access. Derived classes provide the
+ * low-level read operations, e.g. from a local file system or from a web server. The RRawFile is used for non-root
```

Comment:
Is that the intended finally use or a transient one?  i.e. how much of this code would be duplicated in the RFile and its friends?

The intent here seems to be to enforce reading to be done per 'block size' rather than 'buffering' (where the lack of flexibility of the buffering scheme would be an hindrance (See TFileReadCache vs TTreeCache)

Reply:
Right, the intent is to read in reasonably large chunks, in particular for the remote I/O. For instance, the current code in `RSqliteDS` reads in 4k chunks from HTTP servers.

I have no plans making `RRawFile` part of the call stack for ROOT files.  In any case, with `RRawFile::ROptions::fBlockSize == 0`, one can turn off the buffering and possibly apply caching on a higher level.